netutil/netutil.go

```diff
@@ -0,0 +1,129 @@
+package netutil
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+func ipForAddr(addr net.Addr) (net.IP, bool) {
+	switch a := addr.(type) {
+	case *net.IPAddr:
+		return a.IP, true
+	case *net.IPNet:
+		return a.IP, true
+	default:
+		return net.IPv4zero, false
+	}
+}
+
+func PrivateNetworkInterfaces() []string {
```

Comment:
Can we test it? You could have `privateNetworkInterfaces(all []net.Interface)`, test it and have `PrivateNetworkInterfaces()` just calling it.